frappe/model/base_document.py

```diff
@@ -20,6 +21,30 @@
 
 DOCTYPES_FOR_DOCTYPE = ('DocType', 'DocField', 'DocPerm', 'DocType Action', 'DocType Link')
 
+
+class DocStatus(int):
+	def is_draft(self):
+		return self == self.draft()
+
+	def is_submitted(self):
+		return self == self.submitted()
+
+	def is_cancelled(self):
+		return self == self.cancelled()
+
+	@classmethod
+	def draft(cls):
+		return cls(0)
+
+	@classmethod
+	def submitted(cls):
+		return cls(1)
+
+	@classmethod
+	def cancelled(cls):
+		return cls(2)
```

Comment:
is it possible/better to do this?:

```python
class DocStatus(int):
    draft = 0
    submitted = 1
    canceled = 2

	def is_draft(self):
		return self == self.draft
    ....
```

IMO it's better than class methods where you need to call them to get a value for something that's actually pretty static 😅

Reply:
Yes, it's possible, but `DocStatus.draft.is_draft()` wouldn't work, which would be somewhat confusing.

Reply:
> but DocStatus.draft.is_draft() wouldn't work

why? DocStatus should still be comparable to integer 🤔

Reply:
```
>>> class DocStatus(int):
...     draft = 0
...     def is_draft(self):
...             return self == self.draft
... 
>>> DocStatus.draft.is_draft()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'int' object has no attribute 'is_draft'
```